api/src/test/java/org/apache/iceberg/variants/TestSerializedArray.java

```diff
@@ -100,16 +100,22 @@ public void testStringDifferentLengths() {
     assertThat(array.numElements()).isEqualTo(6);
     assertThat(array.get(0).type()).isEqualTo(PhysicalType.STRING);
     assertThat(array.get(0).asPrimitive().get()).isEqualTo("a");
+    assertThat(array.get(0).asPrimitive().sizeInBytes()).isEqualTo(2);
```

Comment:
A few interesting things we can do here, Checking the size is great but we can also assert that the class here is a 
SerializedShortString.

May make sense to have something like
```java
assertVariantString(VariantValue value, String expected) {
   assert physical type
   assert contents equal
   if expected.length < maxShortString
      assert class is SerilaizedShortString
      assert length 1 + expected.length
   else
      assert class is SerializedPrimitive
      assert length 5 + expected.length
}
```